lsp-types/src/Language/LSP/Types/WorkspaceEdit.hs

```diff
@@ -38,12 +40,109 @@ deriveJSON lspOptions ''TextDocumentEdit
 
 -- ---------------------------------------------------------------------
 
+-- | For tagging `CreateFile`/`RenameFile`/`DeleteFile`
+-- Should this be merged with `ResourceOperationKind` ?
+data FileResourceChangeKind
+  = FileResourceChangeCreate
+  | FileResourceChangeRename
+  | FileResourceChangeDelete 
+  deriving (Read, Show, Eq)
+  
+instance ToJSON FileResourceChangeKind where
+  toJSON FileResourceChangeCreate = String "create"
+  toJSON FileResourceChangeRename = String "rename"
+  toJSON FileResourceChangeDelete = String "delete"
+
+instance FromJSON FileResourceChangeKind where
+  parseJSON (String "create") = pure FileResourceChangeCreate
+  parseJSON (String "rename") = pure FileResourceChangeRename
+  parseJSON (String "delete") = pure FileResourceChangeDelete
+  parseJSON _                 = mempty
+
+-- | Options to create a file.
+data CreateFileOptions =
+  CreateFileOptions
+    { -- | Overwrite existing file. Overwrite wins over `ignoreIfExists`
+      _overwrite      :: Bool
+      -- | Ignore if exists.
+    , _ignoreIfExists :: Bool
```

Comment:
To match the specification these should be maybes:

```typescript
/**
 * Options to create a file.
 */
export interface CreateFileOptions {
	/**
	 * Overwrite existing file. Overwrite wins over `ignoreIfExists`
	 */
	overwrite?: boolean;
	/**
	 * Ignore if exists.
	 */
	ignoreIfExists?: boolean;
}
```

```suggestion
    { -- | Overwrite existing file. Overwrite wins over `ignoreIfExists`
      _overwrite      :: Maybe Bool
      -- | Ignore if exists.
    , _ignoreIfExists :: Maybe Bool
```

Reply:
oopsy!